projects/vdk-core/src/vdk/internal/builtin_plugins/job_secrets/base_secrets_impl.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+
+def check_valid_secret(k: str, v: str, supported_types=None) -> None:
```

Comment:
is the only difference between this and check_valid_property the logging?

Reply:
Yes, as Antoni suggested, we can try to refactor this, however I would first like to get it running as I had similar intentions in the Control Service, which ultimately didn't come to fruition as the differences were far too many to extract sensible common classes.